cmd/build/version.go

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package build
+
+var (
+	// These vars will be set by goreleaser.
```

Comment:
It was moved into it's own package so that the variables can be exported for use in `cmd.Run()`. Since testscript requires the main point of entry into the program to have a signature of `func main() int`, I refactored the call to `cmd.Run()` to `internal/cmd/run` and moved the build variables outside `main` package so that it can be exported. I was trying to figure out how to configure GoReleaser to pick up the new location, so I am open for suggestions on how to do that or how to refactor this in a better way.

Reply:
Okay, so my suggestion here is:

- Move `cmd/build` directory to `cmd/tfplugindocs/build` (we're not exposing a "build" CLI 😉 )
- Fix `internal/cmd` for the new import location
- Update `.goreleaser.yaml` with the following for `ldflags`:

```
- '-s -w -X github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs/build.version={{.Version}} -X github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs/build.commit={{.Commit}}'
```

Not sure why it wants the full import path for ldflags 🤷 but it does work when running this locally:

```
goreleaser release --clean --snapshot
./dist/tfplugindocs_darwin_arm64/tfplugindocs --version
tfplugindocs Version 0.16.0-SNAPSHOT-5085dd7 from commit 5085dd7711ef9347228626d493d0fa4d2d7f0010
```